lib/api/include/irods/delay_rule_claim_remove.h

```diff
@@ -0,0 +1,77 @@
+#ifndef IRODS_DELAY_RULE_CLAIM_REMOVE_H
+#define IRODS_DELAY_RULE_CLAIM_REMOVE_H
+
+/// \file
+
+#include "irods/objInfo.h"
+
+struct RcComm;
+
+/// The input data type used to invoke #rc_delay_rule_claim_remove.
+///
+/// \since 5.0.0
+typedef struct DelayRuleClaimRemoveInput
+{
+    /// The catalog ID of the delay rule to claim.
+    ///
+    /// Must be a non-empty string.
+    ///
+    /// \since 5.0.0
+    char rule_id[32]; // NOLINT(cppcoreguidelines-avoid-c-arrays, modernize-avoid-c-arrays)
+
+    /// The set of key-value pair strings used to influence the behavior of the API
+    /// operation.
+    ///
+    /// \since 5.0.0
+    struct KeyValPair cond_input;
+} delayRuleClaimRemoveInp_t;
+
+// NOLINTNEXTLINE(cppcoreguidelines-macro-usage)
+#define DelayRuleClaimRemoveInput_PI "str rule_id[32]; struct KeyValPair_PI;"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/// Removes the delay server claim information from a delay rule if it exists.
+///
+/// Requires \p rodsadmin level privileges.
+///
+/// On success, the following information will be removed from the target delay rule's catalog
+/// entry:
+/// - The hostname or IP of the delay server
+/// - The PID of the delay server
+/// - The time the entry was claimed
+///
+/// \param[in] _comm  A pointer to a RcComm.
+/// \param[in] _input A pointer to a DelayRuleClaimRemoveInput.
+///
+/// \return An integer.
+/// \retval 0  On success.
+/// \retval <0 On failure.
+///
+/// \b Example
+/// \code{.cpp}
+/// RcComm* comm = // Our iRODS connection representing a rodsadmin user.
+///
+/// // Configure the input object for the API call.
+/// struct DelayRuleClaimRemoveInput input;
+/// memset(&input, 0, sizeof(struct DelayRuleClaimRemoveInput));
+///
+/// strncpy(input.rule_id, "12345", sizeof(DelayRuleClaimRemoveInput::rule_id) - 1);
+///
+/// const int ec = rc_delay_rule_claim_remove(comm, &input);
+///
+/// if (ec < 0) {
+///     // Handle error.
+/// }
+/// \endcode
+///
+/// \since 5.0.0
+int rc_delay_rule_claim_remove(struct RcComm* _comm, struct DelayRuleClaimRemoveInput* _input);
```

Comment:
This API allows admins to clear the claim information for a delay rule.

This ability should be exposed through `iadmin`.

That has led me to believe this API should accept a list of rule IDs instead of one. That would allow admins and tools to remove multiple claims without having to cross the network for each one.

Yeah, that is the appropriate enhancement for this API.

Need a test for these APIs too.

Reply:
and with an atomic database update, it should not ever have to communicate partial success of a list of rule IDs to clear...

Reply:
Correct.

Reply:
This seems like a potential candidate for the batch interface of nanodbc. Could prove more efficient than calling a prepared statement multiple times.

Reply:
Partial success is off the table because the ODBC drivers aren't guaranteed to return the correct number of affected rows.

I've updated the API to update a list of delay rules atomically. If one fails, they are all rolled back.